### PR TITLE
Require go 1.23 for build

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### Which issue(s) this PR fixes:
Using the latest golang version for building CRI-O `main`.

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Require go 1.23 to build CRI-O.
```

### CODE REVIEW

**Comment by kannon92** [comment]
@saschagrunert this seems to block quite a few dependencies updates. Anyway to proceed with this?

**Comment by saschagrunert** [comment]
> @saschagrunert this seems to block quite a few dependencies updates. Anyway to proceed with this?
> 
> #8664

We need to wait for go 1.23 to land in OpenShift CI. I expect this in November.

File: go.mod [hunk 1]
```diff
@@ -104,7 +102,7 @@ require (
 	github.com/containerd/cgroups/v3 v3.0.3 // indirect
 	github.com/containerd/console v1.0.4 // indirect
 	github.com/containerd/errdefs v0.1.0 // indirect
-	github.com/containerd/go-runc v1.1.0 // indirect
+	github.com/containerd/go-runc v1.0.0 // indirect
```
Comment (kwilczynski) [inline go.mod:105]: A downgrade?

File: hack/build-rpms.sh [hunk 2]
```diff
@@ -32,13 +32,18 @@ cp -r "${ci_data}/." "${rpm_tmp_dir}/SOURCES"
...
+    os::log::info "Disabling IPv6 for yum ..."
+    echo 'ip_resolve=4' >>/etc/yum.conf
 fi
 # Just in case builddep isn't an installed plugin
 dnf -y install 'dnf-command(builddep)'
 dnf builddep -y "${OS_RPM_SPECFILE}" || true
 
+# Ensure to use latest golang
+GO_VERSION=$(curl -sSfL "https://go.dev/VERSION?m=text" | head -n1)
+curl -sSfL -o- "https://go.dev/dl/$GO_VERSION.linux-amd64.tar.gz" | tar xfz - -C /usr/local
```
Comment (kwilczynski) [inline hack/build-rpms.sh:44]:
```suggestion
curl -sSfL -o- "https://go.dev/dl/${GO_VERSION}.linux-amd64.tar.gz" | tar xfz - -C /usr/local
```

File: go.mod (hunk 1 above)
Comment (saschagrunert) [reply go.mod]: I reverted that. :+1:

File: hack/build-rpms.sh (hunk 2 above)
Comment (saschagrunert) [reply hack/build-rpms.sh]: Fixed as suggested :+1:

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,4 @@
-go 1.22.0
-
-toolchain go1.22.2
```
Comment (kwilczynski) [inline go.mod:3]: I wonder if we should drop this directive. Seems this can cause issues...

**Comment by saschagrunert** [comment]
/override ci/prow/e2e-gcp-ovn

Failures are unrelated.

**Pull request merged**